pipeline/pipeline.go

```diff
@@ -474,9 +514,34 @@ func (p *Pipeline) expandProcs() {
 	p.procCount.Swap(to)
 }
 
+// DeltaWrapper acts as a wrapper around int64
+// and returns the difference between
+// the new and the old value when a new value is set.
+// This is useful for metrics (see the maintenance function).
+type DeltaWrapper struct {
+	val int64
+}
+
+func newDeltaWrapper() *DeltaWrapper {
+	return &DeltaWrapper{0}
+}
+
+func (c *DeltaWrapper) setAndGetDiff(newVal int64) int64 {
+	diff := newVal - c.val
+	c.val = newVal
+	return diff
```

Comment:
`return float64(diff)` to avoid afterward in: `getCounter(outputEventsCountMetric).Add(float64(deltaOutputEvents))`

Reply:
okay, done